src/rules/staticThisRule.ts

```diff
@@ -0,0 +1,127 @@
+/**
+ * @license
+ * Copyright 2019 Palantir Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as utils from "tsutils";
+import * as ts from "typescript";
+
+import * as Lint from "../index";
+import { codeExamples } from "./code-examples/staticThis.examples";
+
+export class Rule extends Lint.Rules.AbstractRule {
+    /* tslint:disable:object-literal-sort-keys */
+    public static metadata: Lint.IRuleMetadata = {
+        ruleName: "static-this",
+        description: "Ban the use of `this` in static methods.",
+        hasFix: true,
+        options: null,
+        optionsDescription: "",
+        optionExamples: [true],
+        /* tslint:disable:max-line-length */
+        rationale: Lint.Utils.dedent`
+            Static \`this\` usage can be confusing for newcomers.
+            It can also become imprecise when used with extended classes when a static \`this\` of a parent class no longer specifically refers to the parent class.
+        `,
+        /* tslint:enable:max-line-length */
+        type: "functionality",
+        typescriptOnly: false,
+        codeExamples,
+    };
+    /* tslint:enable:object-literal-sort-keys */
+
+    public static FAILURE_STRING = (className?: string) => {
```

Comment:
I'm not sold on this set of error strings. "could be a potential reason of runtime errors" could describe most non-formatting rules in TSLint. Additionally, it's a bit redundant to change the error message to the auto-fix, since most users will be able to access that information by applying the auto-fix.

How about <code>Use the parent class name instead of \`this\` when in a \`static\` context.</code>?

Reply:
Fixed